packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -364,11 +365,15 @@ export class DocumentDeltaConnection
             getMaxInternalSocketReconnectionAttempts() + 1;
 
         this._details = await new Promise<IConnected>((resolve, reject) => {
-            const fail = (socketProtocolError: boolean, err: IAnyDriverError) => {
-                this.disposeCore(socketProtocolError, err);
+            const failAndDisposeSocket = (err: IAnyDriverError) => {
+                this.disposeSocket(err);
                 reject(err);
             };
 
+            const fail = (err: IAnyDriverError) => {
```

Comment:
Maybe failConnection? Assuming "connection" associates with connect success/failure messages and session that it creates